OLS.ecl

```diff
@@ -0,0 +1,807 @@
+                                                PROJECT(model, modelToBetas(LEFT));
+  /**
+    * Return raw Beta values as numeric fields
+    *
+    * Extracts Beta values from the model.  Can be used during training and prediction 
+    * phases.  For use during training phase, the 'model' parameter can be omitted.
+    * GetModel will be called to retrieve the model based on the training data.  For
+    * use during prediction phase, a previously persisted model should be provided.
+    * The 'number' field of the returned NumericField records is not used and is set to 1.
+    * The 'id' field of the returned record indicates the position of the Beta
+    * value.  1 provides the Beta for the constant term (i.e. the Y intercept) while
+    * subsequent values reflect the Beta for each correspondingly numbered X feature.
+    * Feature 1 corresponds to Beta with 'id' = 2 and so on.
+    * If 'model' contains multiple work-items, Separate sets of Betas will be returned
+    * for each of the 'myriad' models (distinguished by 'wi').
+    *
+    * @param model Optional parameter provides a model that was previously retrieved
+    *               using GetModel.  If omitted, GetModel will be used as the model.
+    * @return DATASET(NumericField) containing the Beta values.
+    * 
+    */
+  EXPORT DATASET(NumericField) Betas(DATASET(Layout_Model) model=GetModel) := FUNCTION
+    nfBetas := PbbConverted.MatrixToNF(mBetas(model));
+    // Return betas distributed by work item so that the analytics will be balanced across
+    // nodes.
+    return DISTRIBUTE(nfBetas, wi);
+  END;
+   
+  /**
+    * Predict the dependent variable values (Y) for any set of independent
+    * variables (X).  Returns a predicted Y value for each observation
+    * (i.e. record) of X.
+    * 
+    * This supports the 'myriad' style interface in that multiple independent
+    * work items may be present in 'newX', and multiple independent models may
+    * be provided in 'model'.  The resulting predicted values will also be
+    * separable by work item (i.e. wi).
+    *
+    * @param newX   The set of observations if independent variables in 
+    *               DATASET(NumericField) format.
+    * @param model  Optional.  A model that was previously returned from GetModel (above).
+    *               Note that a model from a previous run will only be valid
+    *               if the field numbers in X are the same as when the model
+    *               was learned.  If this parameter is omitted, the current
+    *               model will be used.
+    * @return       An estimation of the corresponding Y value for each
+    *               observation of newX.  Returned in DATASET(NumericField)
+    *               format with field number (i.e. 'number') always = 1.
+    */
+  EXPORT DATASET(NumericField) Predict(DATASET(NumericField) newX, 
+  									DATASET(Layout_Model) model=GetModel) := FUNCTION
```

Comment:
This looks like you allowed tab characters.  Please replace.  The code really should have good indentation even when viewed in a browser.